### DISCUSSION TITLE
How to get fields information from stdin ?

### DISCUSSION THREAD

**Discussion by kpym** [mithrandie/csvq#70] [Q&A] [answered]
I'm trying something like `cat mydata.csv | csvq fields` but I can't get the desired result. Is there an option to indicate that the file is `stdin`?

**Comment by ondohotola** [comment]
RTFM

EL

—

**Reply by kpym** [reply]
@ondohotola I looked in the manual but I couldn't find how to do it (if it's possible). It seems that `csvq fields` requires a filename and that `stdin` is not supported.

**Comment by ondohotola** [comment]
Look at the the SELECT clause.

el

On 06/12/2021 12:49, Kroum Tzanev wrote:
> @ondohotola I looked in the manual but I couldn't find how to do it
> (if it's possible).  It seems that |csvq fields| requires a filename
> and that |stdin| is not supported.
[...]

-- 
Dr. Eberhard W. Lisse   \         /       Obstetrician & Gynaecologist
***@***.***             / *      |  Telephone: +264 81 124 6733 (cell)
PO Box 8421 Bachbrecht  \      /  If this email is signed with GPG/PGP
10007, Namibia           ;____/ Sect 20 of Act No. 4 of 2019 may apply
	 Book your Covid-19 Vaccination at https://c19.idtoday.com.na

**Reply by kpym** [reply]
@ondohotola I know that in the `select` clause I can use `...from stdin`, but my question is specifically about `fields` subcommand.

**Comment by ondohotola** [comment]
You'll probably have to generate a temporary file and read that.

el

**Reply by kpym** [reply]
@ondohotola That's not what I was asking. I know I can generate a temporary file. After your kind "RTFM", I was hoping you knew how to read from stdin for the `fields` subcommand.

**Comment by ondohotola** [comment]
The manual quite clearly if short, in my view and from retrospect, shows
that a file name is required.

On 06/12/2021 14:09, Kroum Tzanev wrote:
> @ondohotola <https://github.com/ondohotola> That's not what I was
> asking.  I know I can generate a temporary file.  After your kind
> "RTFM", I was hoping you knew how to read from stdin for the |fields|
> subcommand.

-- 
Dr. Eberhard W. Lisse   \         /       Obstetrician & Gynaecologist
***@***.***             / *      |  Telephone: +264 81 124 6733 (cell)
PO Box 8421 Bachbrecht  \      /  If this email is signed with GPG/PGP
10007, Namibia           ;____/ Sect 20 of Act No. 4 of 2019 may apply
	 Book your Covid-19 Vaccination at https://c19.idtoday.com.na

**Comment by mithrandie** [comment, marked as answer]
The fields subcommand does not have such an option. However, you can use SHOW command as a query.

`cat mydata.csv | csvq ‘SHOW FIELDS FROM STDIN’`

https://mithrandie.github.io/csvq/reference/built-in.html#show_fields

**Reply by kpym** [reply]
@mithrandie Thanks ! This is what I was looking for 😃

**Comment by ondohotola** [comment]
And it's even in the Manual :-)-O

el

On 06/12/2021 14:27, Kroum Tzanev wrote:
> @mithrandie Thanks ! This is what I was looking for 😃
[...]


-- 
Dr. Eberhard W. Lisse   \         /       Obstetrician & Gynaecologist
***@***.***             / *      |  Telephone: +264 81 124 6733 (cell)
PO Box 8421 Bachbrecht  \      /  If this email is signed with GPG/PGP
10007, Namibia           ;____/ Sect 20 of Act No. 4 of 2019 may apply
     Book your Covid-19 Vaccination at https://c19.idtoday.com.na

**Reply by kpym** [reply]
@ondohotola So you should RTFM before to try to answer 😉

**Comment by ondohotola** [comment]
And tell you to RTFM, therafter?

el



On 06/12/2021 16:23, Kroum Tzanev wrote:
> @ondohotola  So you should RTFM before to try to answer 😉
[...]


-- 
Dr. Eberhard W. Lisse   \         /       Obstetrician & Gynaecologist
***@***.***             / *      |  Telephone: +264 81 124 6733 (cell)
PO Box 8421 Bachbrecht  \      /  If this email is signed with GPG/PGP
10007, Namibia           ;____/ Sect 20 of Act No. 4 of 2019 may apply
     Book your Covid-19 Vaccination at https://c19.idtoday.com.na
